core/src/main/java/io/grpc/Codec.java

```diff
@@ -0,0 +1,59 @@
+package io.grpc;
+
+
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Encloses classes related to the compression and decompression of messages.
+ */
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/492")
+public interface Codec extends Compressor, Decompressor {
+  /**
+   * A gzip compressor and decompressor.  In the future this will likely support other
+   * compression methods, such as compression level.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/492") class Gzip implements Codec {
```

Comment:
That actually made me remember some tip to not put constants on Interfaces.  Moved Identity to its own class.